packages/manager/apps/web-ongoing-operations/src/pages/upload/Upload.tsx

```diff
@@ -0,0 +1,264 @@
+import { BaseLayout } from '@ovh-ux/manager-react-components';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useNavigate, useParams } from 'react-router-dom';
+import {
+  OdsButton,
+  OdsDivider,
+  OdsFileUpload,
+  OdsMessage,
+  OdsRadio,
+  OdsText,
+} from '@ovhcloud/ods-components/react';
+import {
+  ODS_BUTTON_VARIANT,
+  ODS_DIVIDER_COLOR,
+  ODS_DIVIDER_SPACING,
+  ODS_MESSAGE_COLOR,
+  ODS_TEXT_PRESET,
+  OdsFile,
+  OdsFileChangeEventDetail,
+  OdsFileRejectedEventDetail,
+  OdsFileUploadCustomEvent,
+} from '@ovhcloud/ods-components';
+import { updateTask } from '@/data/api/web-ongoing-operations';
+import SubHeader from '@/components/SubHeader/SubHeader';
+import { saveFile } from '@/data/api/document';
+import { urls } from '@/routes/routes.constant';
+import Loading from '@/components/Loading/Loading';
+import { useDomain, useDomainArgument } from '@/hooks/data/data';
+import { updateOperationStatus } from '@/data/api/ongoing-operations-actions';
+import { OperationName } from '@/enum/operationName.enum';
+import { ParentEnum } from '@/hooks/useOngoingOperationDatagridColumns';
+
+export default function Upload() {
+  const { t } = useTranslation('dashboard');
+  const { id, type } = useParams<{ id: string; type: string }>();
+  const navigate = useNavigate();
+  const [operationName, setOperationName] = useState<OperationName>(null);
+  const defaultStatus = { label: '', message: '' };
+  const [status, setStatus] = useState(defaultStatus);
+  const [uploadedFiles, setUploadedFiles] = useState<OdsFile[]>([]);
+  const [errorUpload, setErrorUpload] = useState<string>('');
+  const maxFile = 1;
+  const paramId = Number(id);
+
+  const { data: domain, isLoading: domainLoading } = useDomain(paramId);
+
+  const {
+    data: domainArgument,
+    isLoading: argumentLoading,
+  } = useDomainArgument(paramId, type);
+
+  const executeActionOnOperation = async (operationID: number) => {
+    try {
+      await updateOperationStatus(
+        ParentEnum.Domain,
+        operationID,
+        operationName,
+      ).then(() => navigate(urls.domain));
+    } catch {
+      setStatus({
+        label: ODS_MESSAGE_COLOR.warning,
+        message: t('domain_operations_upload_error'),
+      });
+    }
+  };
```

Comment:
I think you can use useMutation from tanstack query you have some example on manager base code.